tokenizers/src/pre_tokenizers/metaspace.rs

```diff
@@ -1,6 +1,45 @@
+use crate::tokenizer::{Decoder, PreTokenizedString, PreTokenizer, Result, SplitDelimiterBehavior};
 use serde::{Deserialize, Deserializer, Serialize};
+use std::fmt;
 
-use crate::tokenizer::{Decoder, PreTokenizedString, PreTokenizer, Result, SplitDelimiterBehavior};
+/// Enum representing options for the metaspace prepending scheme.
+#[derive(Debug, Clone, PartialEq, Serialize, Eq, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum PrependScheme {
+    /// Specifies that the scheme should be prepended only once, on the first split.
+    First,
+    /// Specifies that the space should not be prepended.
+    Never,
+    /// Specifies that the scheme should always be prepended.
+    Always,
+}
+
+impl From<&str> for PrependScheme {
+    fn from(s: &str) -> Self {
+        match s.to_lowercase().as_str() {
+            "first" => PrependScheme::First,
+            "never" => PrependScheme::Never,
+            "always" => PrependScheme::Always,
+            _ => panic!("Invalid value for PrependScheme: {}", s),
```

Comment:
Panicking is NOT okay in transformers. We should NEVER panic since we're a library.

For that there is `TryFrom` which returns `Result<T>`. However I don't think this should be done in the Rust layer but in the Python layer instead.